x-pack/platform/test/spaces_api_integration/read_only_objects/apis/spaces/read_only_objects.ts

```diff
@@ -0,0 +1,1341 @@
+
+        const { cookie: adminCookie } = await loginAsKibanaAdmin();
+        const response = await supertestWithoutAuth
+          .put('/read_only_objects/change_access_mode')
+          .set('kbn-xsrf', 'true')
+          .set('cookie', ownerCookie.cookieString())
+          .send({
+            objects: [{ id: objectId, type: 'read_only_type' }],
+            newAccessMode: 'read_only',
+          })
+          .expect(200);
+        expect(response.body.objects).to.have.length(1);
+        expect(response.body.objects[0].id).to.eql(objectId);
+        expect(response.body.objects[0].type).to.eql('read_only_type');
+
+        const getResponse = await supertestWithoutAuth
+          .get(`/read_only_objects/${objectId}`)
+          .set('kbn-xsrf', 'true')
+          .set('cookie', adminCookie.cookieString())
+          .expect(200);
+
+        expect(getResponse.body.accessControl).to.have.property('accessMode', 'read_only');
+
+        const updateResponse = await supertestWithoutAuth
+          .put('/read_only_objects/update')
+          .set('kbn-xsrf', 'true')
+          .set('cookie', ownerCookie.cookieString())
+          .send({ objectId, type: 'read_only_type' })
+          .expect(200);
+        expect(updateResponse.body.id).to.eql(objectId);
+        expect(updateResponse.body.attributes).to.have.property(
+          'description',
+          'updated description'
+        );
+      });
+
+      it('should throw when trying to change access mode on locked objects when not owner', async () => {
+        const { cookie: ownerCookie, profileUid } = await loginAsObjectOwner(
+          'test_user',
+          'changeme'
+        );
+        const createResponse = await supertestWithoutAuth
+          .post('/read_only_objects/create')
+          .set('kbn-xsrf', 'true')
+          .set('cookie', ownerCookie.cookieString())
+          .send({ type: 'read_only_type', isReadOnly: true })
+          .expect(200);
+
+        const objectId = createResponse.body.id;
+        expect(createResponse.body.accessControl).to.have.property('owner', profileUid);
+
+        await activateSimpleUserProfile();
+        const { cookie: notOwnerCookie } = await loginAsNotObjectOwner('simple_user', 'changeme');
+        const updateResponse = await supertestWithoutAuth
+          .put('/read_only_objects/change_access_mode')
+          .set('kbn-xsrf', 'true')
+          .set('cookie', notOwnerCookie.cookieString())
+          .send({
+            objects: [{ id: objectId, type: 'read_only_type' }],
+            newAccessMode: 'read_only',
+          })
+          .expect(403);
+        expect(updateResponse.body).to.have.property('message');
+        expect(updateResponse.body.message).to.contain(
+          `Access denied: Unable to manage access control for read_only_type`
+        );
+      });
+
+      it('allows updates on removing read only access mode', async () => {
```

Comment:
Nit - might be redundant, but I think it is fine to leave it
```suggestion
      it('allows updates by non-owner after removing read only access mode', async () => {
```